test/rpc/test_rpc_schema.py

```diff
@@ -0,0 +1,320 @@
+    """
+    Test kwargs validation allows valid requests
+    """
+
+    resp = rpc_call("starknet_call", params=params)
+
+    # Error will be raised because address is correctly formatted but incorrect
+    error = resp["error"]
+    assert all(error["code"] != code.value for code in RpcErrorCode)
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+def test_schema_does_not_raise_on_correct_args():
+    """
+    Test args validation allows valid requests
+    """
+
+    resp = rpc_call(
+        "starknet_call",
+        params=[
+            {
+                "contract_address": "0x01",
+                "entry_point_selector": rpc_felt(get_selector_from_name("get_balance")),
+                "calldata": [],
+            },
+            "latest",
+        ],
+    )
+
+    # Error will be raised because address is correctly formatted but incorrect
+    error = resp["error"]
+    assert all(error["code"] != code.value for code in RpcErrorCode)
+
+
+def test_schema_with_optional_values():
+    """
+    Test schema validation allowing omitting non-required values
+    """
+    with patch(
+        "starknet_devnet.blueprints.rpc.schema._request_schemas_for_method", MagicMock()
+    ) as mocked:
+        mocked.return_value = {
+            "key": {
+                "is_required": True,
+                "$ref": "#/components/schemas/STORAGE_KEY",
+                "components": {
+                    "schemas": {
+                        "STORAGE_KEY": {
+                            "type": "string",
+                            "title": "A storage key",
+                            "$comment": "A storage key, represented as a string of hex digits",
+                            "description": "A storage key. Represented as up to 62 hex digits, 3 bits, and 5 leading zeroes.",
+                            "pattern": "^0x0[0-7]{1}[a-fA-F0-9]{0,62}$",
+                        }
+                    }
+                },
+            },
+            "value": {
+                "is_required": False,
+                "$ref": "#/components/schemas/STORAGE_KEY",
+                "components": {
+                    "schemas": {
+                        "STORAGE_KEY": {
+                            "type": "string",
+                            "title": "A storage key",
+                            "$comment": "A storage key, represented as a string of hex digits",
+                            "description": "A storage key. Represented as up to 62 hex digits, 3 bits, and 5 leading zeroes.",
+                            "pattern": "^0x0[0-7]{1}[a-fA-F0-9]{0,62}$",
+                        }
+                    }
+                },
+            },
+        }
+
+        # Testing private methods is not ideal, but I have no other idea of testing this logic
+        # because mocking doesn't work with devnet running in background subprocess
+        params = {"key": "0x01"}
+        _assert_valid_rpc_request(**params, method_name="starknet_method")
```

Comment:
I fail to understand what exactly is tested here and why this test is necessary.

Reply:
`params` of the most RPC methods, alongside with the schema, have a `"required": true` field (no method exist currently with required being false). My schema validation is supposed to support not required params anyways.

This test is meant to verify if this logic does work, you can see `is_required` keys in schemas used in mock. In non-mocked situation these should be added by the schema parser based on value of `"required"` key in the schema.

Reply:
So why did `is_required` have to be introduced if there is `required`?

Reply:
In the `_request_schemas_for_method`, when I "repackage" the json for validation purposes there's a code like
```py
    for param in params_json:
        name = param["name"]
        is_required = param.get("required", False)
        schema = {
            **param["schema"],
            "is_required": is_required,
            "components": {"schemas": schemas},
        }
        request_schemas[name] = schema
```

I can change it to just `required` from `is_required`. Honestly don't know why I chose to rename it. Can't directly use it though, because I'm not sure if for optional parameters json rpc would have `"required": false` or I'd just be omitted entirely.

Reply:
I guess it's ok as it is.